modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -101,6 +102,7 @@ func (h *Handler) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	// to write tests before making any more changes to it
 	upstreamDialAddress := func(upstreamAddr string) (string, error) {
 		var network, scheme, host, port string
+		placeholder := regexp.MustCompile(`\{[[:graph:]]+\}`)
```

Comment:
Why does this logic/check have to be repeated in the Caddyfile adapter?

(If we keep it, this should be brought out into a package-level variable like `var placeholderPattern = ...` or something, then reused...)